python/trigger/tpreplay_application/__main__.py

```diff
@@ -0,0 +1,381 @@
+    logging.debug("TPStream files loaded: %s", tpstream_files)
+
+    return tpstream_files
+
+def check_files(files):
+    """
+    Very basic checks on the provided TPStream files.
+    """
+    for a_file in files:
+        ### Checks
+        # check file exist
+        path = Path(a_file)
+        if not path.exists():
+            logging.error("File %s does not exist!", a_file)
+            sys.exit(1)
+        # check it's hdf5
+        if not path.suffix.lower() in ('.h5', '.hdf5'):
+            logging.error("File %s does not seem to be hdf5 file!", a_file)
+            sys.exit(1)
+
+def extract_rous_and_planes(files: List[str], channel_map, planes_to_filter: Set[int], verbose: bool) -> (List[TPStreamFile], ROUPlaneData):
+    """
+    This function goes over the provided TPStream files.
+    It extracts the readout units used to generate the data in the files.
+    It also extracts unfiltered planes for each readout unit.
+    """
+    logging.info("Extracting ROUs and planes from files")
+    all_tpstream_files = []
+    rou_plane_data = ROUPlaneData()
+
+    valid_subdetectors = {
+        int(detdataformats.DetID.Subdetector.kHD_TPC),
+        int(detdataformats.DetID.Subdetector.kVD_BottomTPC),
+        int(detdataformats.DetID.Subdetector.kVD_TopTPC),
+        int(detdataformats.DetID.Subdetector.kNDLAr_TPC)
+    }
+
+    for tpstream_file in files:
+        logging.debug("Processing file: %s", tpstream_file)
+        loaded_file = HDF5RawDataFile(tpstream_file)
+        # check is tpstream
+        if not loaded_file.is_timeslice_type:
+            logging.error("File %s is not a TP Stream file!", tpstream_file)
+            sys.exit(1)
+        # check has records
+        all_record_ids = loaded_file.get_all_record_ids()
+        if not all_record_ids:
+            logging.error("File %s does not have valid records!", tpstream_file)
+            sys.exit(1)
+        first_record = all_record_ids[0]
+        # check has source IDs
+        source_ids = loaded_file.get_source_ids_for_fragment_type(first_record, "Trigger_Primitive")
+        if len(source_ids) == 0:
+            logging.error("File %s does not have valid SourceIDs!", tpstream_file)
+            sys.exit(1)
+        logging.debug("SIDs: %s", source_ids)
+
+        for i, sid in enumerate(source_ids):
+            frag = loaded_file.get_frag(first_record, sid)
+            # check frag has data
+            if frag.get_data_size() < 1:
+                logging.error("File %s has an empty fragment!", tpstream_file)
+                sys.exit(1)
+            tp = trgdataformats.TriggerPrimitive(frag.get_data(0))
+
+            if i == 0:
+                all_tpstream_files.append(TPStreamFile(tpstream_file, tp.time_start, 0))
+                logging.debug("First time start: %s", tp.time_start)
+
+            # check subdetector
+            subdet = tp.detid
+            print(subdet)
+            if subdet not in valid_subdetectors:
+                continue
```

Comment:
```suggestion
            if tp.detid not in valid_subdetectors:
                continue
```

I think it's that print line that caused many 0s being printed out when I tried running this code. So either remove, or maybe add extra info to the `print()` function so people know what these numbes are :)